src/web/app/components/question-edit-form/question-edit-form.component.ts

```diff
@@ -246,15 +246,25 @@ export class QuestionEditFormComponent implements OnInit {
     if (allowedRecipientTypes.indexOf(recipientType) === -1) {
       newRecipientType = allowedRecipientTypes[0];
     }
-    this.triggerModelChangeBatch({
-      giverType,
-      recipientType: newRecipientType,
-      commonVisibilitySettingName: 'Please select a visibility option',
-      isUsingOtherVisibilitySetting: false,
-      showResponsesTo: [],
-      showGiverNameTo: [],
-      showRecipientNameTo: [],
-    });
+    // do not reset the visibility settings if reverting feedback path to preset template provided
+    if (this.model.isUsingOtherFeedbackPath && this.isCustomFeedbackVisibilitySettingAllowed &&
+        this.model.giverType === giverType && this.model.recipientType === newRecipientType) {
+      this.triggerModelChangeBatch({
+        giverType,
+        isUsingOtherFeedbackPath: false,
+        recipientType: newRecipientType,
```

Comment:
Do you need to trigger an update of `recipientType` if it already is `newRecipientType` by the condition above? Likewise for `giverType`.
The update here following the equality checks seems redundant.